### ISSUE TITLE
Firefox Support Plz

### ISSUE THREAD

**Issue by SandNerd** [wakatime/chrome-wakatime#49]
[WebExtensions APIs](https://github.com/bfred-it/Awesome-WebExtensions) are pretty similar to chrome and a little bit tweaking should be sufficient to make firefoxers happy.

**Comment by Eeems** [comment]
I'm working on getting a pull request together on [my fork](https://github.com/Eeems/chrome-wakatime)

**Comment by Eeems** [comment]
Pull request #75 will need to be merged to have proper support. That said, it will be up to @alanhamlett to upload an official addon to the Mozilla addons site.

**Comment by kovetskiy** [comment]
hi guys, what is the status of the issue? I see those PRs are merged

**Comment by Eeems** [comment]
@kovetskiy Still waiting on @alanhamlett to build and upload an official addon to the Mozilla addons site.

**Comment by Xunnamius** [comment]
Any word on this? I recently switched to Firefox from Chrome.

**Comment by Eeems** [comment]
@Xunnamius Nope, still waiting on @alanhamlett

**Comment by alanhamlett** [comment]
I'm wanting to get the Chrome version out of beta first (Chrome's version shouldn't even be 1.0 yet since it's still beta). That means fixing some bugs, implementing domain-based rules for smarter tracking, and a few other features. Then we'll release both Chrome and Firefox plugins with a 1.0 version.

**Comment by kovetskiy** [comment]
is there any ETA and/or roadmap?

**Comment by alanhamlett** [comment]
> any ETA and/or roadmap?

No, unfortunately it's not on our roadmap yet since we haven't scoped out the work. It's just on our backlog with no ETA. I can say it's next on the list after the Desktop app.

**Comment by RazgrizHsu** [comment]
support firefox please

**Comment by NeckBeardPrince** [comment]
Guys i feel like we should have a Firefox extension. I mean you have this [page](https://wakatime.com/firefox) but you click on it and you get "It's not ready". Come to find out it isn't even in the works?

**Comment by Eeems** [comment]
I mean, the work is mostly done. I've got a build of it that works. @alanhamlett just needs to actually publish it and do some cleanup.

**Comment by 4face-studi0** [comment]
Is somebody black-mailing you to do not publish it? :D

**Comment by noomz** [comment]
My development time is mostly on Firefox. It's a waste of time guessing missing time records. It's too long waiting for the plugin.

**Comment by alanhamlett** [comment]
#75 was merged to add Firefox support.

**Comment by Eeems** [comment]
@alanhamlett While the codebase supports Firefox, you have yet to publish the extension for people to download.

**Comment by llinfeng** [comment]
@alanhamlett Could you point me to a working branch/commit where the Firefox add-on is usable? I see the following after the following steps. (I have no idea how `npm` works. Excuse me for posting the extensive set of steps.)
```
# 1. clone this repo
# 2. cd to the folder
# 3. get `npm` ready: otherwise, I get `clap not found` error
npm install
# 4. build the extension locally
npm start
# 5. Look for a zip file in: `/WakatimeBuild/chrome-wakatime/dist/firefox`
# 6. Open Firefox and visit: `about:debugging#/runtime/this-firefox` 
       # ==> this got me to "install" the unsigned add-on
```

Well, after these steps, I got slammed into a wall :) ==> no way to login, nor apply any white/black list.

![image](https://user-images.githubusercontent.com/5890407/108526643-61234500-729f-11eb-8323-4e1b287dd3ed.png)

**Comment by NeckBeardPrince** [comment]
@llinfeng  do yourself a favor and find another platform other than Wakatime. There is clearly zero importance to this issue for the owner/maintainer. This has been ignored for far too long. They merged the PR to support Firefox yet, they haven't built the extension and released it.

**Comment by llinfeng** [comment]
@NeckBeardPrince I suppose WakaTime is mostly a one-man operation? I am personally amazed by the full set of editor support it provides. For now, I suppose we need some final touch from @alanhamlett?

RE: support for Firefox - I think we are close to at least getting something to work: for now, the `chrome` and `firefox` extensions look rather identical. Hopefully, they shall function in a similar manner? I was surprised to find two folders populated under `chrome-wakatime/dist/` directory. Except for the "Options go here" placeholder, I would imagine one can get a working Firefox extension after building and singing the add-on himself/herself? 

----

This reminds me of the dark time when the [vimperator](http://vimperator.org/) project was discontinued, where I was using a self-compiled and self-signed add-on on an old Firefox ESR version until that ESR version is discontinued in 2018/2019? I can live with Wakatime should it get rid of the "Options go here" sign and function properly: Firefox support is rather helpful, in particular as I find a performance improvement on Firefox when accessing a local server for Jupyterlab notebooks.

**Comment by NeckBeardPrince** [comment]
> @NeckBeardPrince I suppose WakaTime is mostly a one-man operation? I am personally amazed by the full set of editor support it provides. For now, I suppose we need some final touch from @alanhamlett?
> 
> RE: support for Firefox - I think we are close to at least getting something to work: for now, the `chrome` and `firefox` extensions look rather identical. Hopefully, they shall function in a similar manner? I was surprised to find two folders populated under `chrome-wakatime/dist/` directory. Except for the "Options go here" placeholder, I would imagine one can get a working Firefox extension after building and singing the add-on himself/herself?
> 
> This reminds me of the dark time when the [vimperator](http://vimperator.org/) project was discontinued, where I was using a self-compiled and self-signed add-on on an old Firefox ESR version until that ESR version is discontinued in 2018/2019? I can live with Wakatime should it get rid of the "Options go here" sign and function properly: Firefox support is rather helpful, in particular as I find a performance improvement on Firefox when accessing a local server for Jupyterlab notebooks.

According to [this](https://www.linkedin.com/company/wakatime) there are 4.

**Comment by RyanMulready** [comment]
For anyone who wonders in via google like me wondering what the state here is a the solution that worked for me.

1. `git clone` the repo
2. `npm install`
3. `npm run start`
4. in firefox `about:debugging#/runtime/this-firefox`
5. Select `Load Temporary Addon` and select the `mainfest.json` file in the ***ROOT*** directory

The files created under `dist/` under the "new" build does not work and is not ready.

**Comment by Eeems** [comment]
> The files created under `dist/` under the "new" build does not work and is not ready.

:( Any ideas on what's missing at this point for the build?

**Comment by RyanMulready** [comment]
> > The files created under `dist/` under the "new" build does not work and is not ready.
> 
> :( Any ideas on what's missing at this point for the build?

No sorry, it looked liked the react instance was failing to mount which is why you get `OPTIONS GO HERE` and not your component. I didn't go any deeper than that.

**Comment by NeckBeardPrince** [comment]
Christ, this still hasn't happened yet?

**Comment by jvelezpo** [comment]
FF support is ready 😎 
Closing this 
* Version 3.0.4 already ships with support for FF
* It is already released in FF add on official website https://addons.mozilla.org/en-US/firefox/addon/wakatimes/

**Comment by Freed-Wu** [comment]
<https://wakatime.com/firefox> should be updated due to <https://addons.mozilla.org/en-US/firefox/addon/wakatimes/>

**Comment by Freed-Wu** [comment]
Firefox display unknown host name. Is it a bug? Can chrome display host name correctly?
![Screenshot from 2023-11-28 02-29-59](https://github.com/wakatime/browser-wakatime/assets/32936898/08bbd9f0-ce42-47be-8c1e-645b03be49c2)

**Comment by Eeems** [comment]
> Firefox display unknown host name. Is it a bug? Can chrome display host name correctly? ![Screenshot from 2023-11-28 02-29-59](https://private-user-images.githubusercontent.com/32936898/285971971-08bbd9f0-ce42-47be-8c1e-645b03be49c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExMTA4MDcsIm5iZiI6MTcwMTExMDUwNywicGF0aCI6Ii8zMjkzNjg5OC8yODU5NzE5NzEtMDhiYmQ5ZjAtY2U0Mi00N2JlLThjMWUtNjQ1YjAzYmU0OWMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI3VDE4NDE0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmMDM2NTU0NmFiNGZjYWE2NDZiYmFkNTY3MWVhMjMyODQ1ZDE0MjczMWM4ZDA4NjZiZDliZWM0MThlMTE5ZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.b21gEf2hZF3cnwKC64q7NpWq3-SFE9Q-CkV6G8edeuI)

I think you meant to open a new issue instead of adding a comment to an old closed issue.
